python/lsst/pex/config/config.py

```diff
@@ -237,23 +237,26 @@ def save(self, outfile, instance):
         """
         value = self.__get__(instance)
         fullname = _joinNamePath(instance._name, self.name)
+
+        # write full documentation string as comment lines (i.e. first character is #)
+        doc = "# " + re.sub(r'\n', r'\n# ', str(self.doc))
         if isinstance(value, float) and (math.isinf(value) or math.isnan(value)):
             # non-finite numbers need special care
-            print >> outfile, "%s=float('%r')"%(fullname, value)
+            print >> outfile, "%s\n%s=float('%r')\n"%(doc, fullname, value)
```

Comment:
The Butler has been criticized for always warning that optional configuration overrides are not available; I'm not sure why a configuration override file should do the same thing.

Passing a log instance to the configuration override file evaluator could be an alternative.

If configuration override files can perform I/O (or computation), we should set very clear rules on what kinds of I/O (or computation) are permissible.  Doing any I/O or computation likely means that there is no way to understand what a configuration override file will be doing (e.g. in terms of imports) without just executing it; I think we've already passed that point.  But we should be able to forbid `import scipy` or `open(".config.tmp", "w")` or anything else that might make full provenance hard to capture.